com.unity.render-pipelines.universal/Runtime/UniversalRenderer.cs

```diff
@@ -475,21 +475,40 @@ public override void Setup(ScriptableRenderContext context, ref RenderingData re
                 EnqueuePass(m_XROcclusionMeshPass);
 #endif
 
+            bool requiresCopyColorPass = renderingData.cameraData.requiresOpaqueTexture || renderPassInputs.requiresColorTexture;
+            // If a depth texture was created we necessarily need to copy it, otherwise we could have render it to a renderbuffer.
+            bool requiresDepthCopyPass = !requiresDepthPrepass
+                && renderingData.cameraData.requiresDepthTexture
+                && createDepthTexture;
+
             if (this.actualRenderingMode == RenderingMode.Deferred)
                 EnqueueDeferred(ref renderingData, requiresDepthPrepass, renderPassInputs.requiresNormalsTexture, mainLightShadows, additionalLightShadows);
             else
+            {
+                RenderBufferStoreAction opaquePassColorStoreAction = RenderBufferStoreAction.Store;
+                RenderBufferStoreAction opaquePassDepthStoreAction = RenderBufferStoreAction.Store;
```

Comment:
i'd encapsulate this whole part with `if cameraTargetDescriptor.msaaSamples > 1`, as otherwise nothing changes and that would reduce some clutter.

or even do something like `RenderBufferStoreAction opaquePassColorStoreAction =  ameraTargetDescriptor.msaaSamples > 1 ? ((requiresCopyColorPass || requiresDepthCopyPass) ? RenderBufferStoreAction.StoreAndResolve : RenderBufferStoreAction.Resolve) : RenderBufferStoreAction.Store` which isn't as readable, but reduces boilerplate code